static/js/pm_list.js

```diff
@@ -110,10 +113,15 @@ exports._build_private_messages_list = function (active_conversation, max_privat
     return recipients_dom;
 };
 
-exports.rebuild_recent = function (active_conversation) {
+exports.rebuild_recent = function (active_conversation, new_message_arrived) {
     remove_expanded_private_messages();
     if (private_messages_open) {
-        var max_private_messages = 5;
+        var max_private_messages;
+        if (new_message_arrived && private_more_messages_open) {
```

Comment:
If a new message arrives, it should be at the top of the list, so I'm not sure why we would extend the list beyond 5 messages in this case.